### ISSUE TITLE
[Question] How to use minor version but monitor for patch

### ISSUE THREAD

**Issue by RichyHBM** [fmartinou/whats-up-docker#395]
Hello, I am looking to answer a question regarding docker image versions.

I want to use minor versions for my docker images where available, i.e. 1.23, but have WUD notify me when a new patch version is available. This works ok using patch compatible regex, however if I update that container to the latest patch, WUD still lists it as an available update, even though they are at the latest.

Is there any setting that can achieve the desired outcome? Would I need to watch digests as I currently disable this where possible to minimize on dockerhub API calls

**Comment by fmartinou** [comment]
Hello,

I'm not sure to exactly understand your situation; can you provide a small example of your configuration, please?


If I take the example of [nginx](https://hub.docker.com/_/nginx), they provide the same image as `1`, `1.27` and `1.27.0`.

I basically would say:
- if you want to keep your image up to date as soon as a new patch is available, use the `1.27.0` tag and set wud the include regex `\d\.\d\.\d`
- if you want to keep your image up to date as soon as a new minor is available, use the `1.27` tag and set wud the include regex `\d\.\d`
- if you want to keep your image up to date as soon as a new major is available, use the `1` tag and set wud the include regex `\d`

But maybe I have misunderstood your issue...

**Comment by rheicide** [comment]
@fmartinou I think I get what @RichyHBM meant, as apparently I'm in a similar situation. Let me explain.

- Let's say I'm running a Postgres container; in my `compose.yml` I specify the image as `postgres:16`.
- I want WUD to notify me when a new minor version like `16.3` is available (which means `16` has also been updated on Docker Hub and I can just run `docker compose pull` to get it without having to change `compose.yml`), so I set `wud.tag.include=^16\.\d+$$`.
- Now, even if my local `postgres:16` image is up-to-date and actually has the same digest as `postgres:16.3`'s, WUD still shows that a minor update is available.
- Thinking that if WUD sees that they have the same digest, it'll know that there's in fact no update, I set `wud.watch.digest=true` and recreate the container.
- However, WUD still says that there's an update, even though now I can see on WUD's container page that the current image's digest and the "new" one's digest are the same:
  ![image](https://github.com/fmartinou/whats-up-docker/assets/953254/58778089-574f-4b05-b000-3c9db99e2ae5)
  ![image](https://github.com/fmartinou/whats-up-docker/assets/953254/05997b16-202f-4e5d-af1e-697a6b3b4246)

If WUD would compare digests with `wud.watch.digest=true` even when tags are different (`16` vs. `16.3` in this example) to decide whether there's an update, it would be perfect.

I'm aware that I can set `wud.tag.include=^16$$` and `wud.watch.digest=true` to get "correct" notifications, but that way WUD won't be able to show what minor version the update is—is it `16.3` or `16.4`?

I think this is kinda finicky and it's totally fine if you decide to keep the current behavior. Just wanted to chime in and make it clear. Thanks for this useful piece of software!
